stubs/wurlitzer/wurlitzer.pyi

```diff
@@ -0,0 +1,97 @@
+import contextlib
+import io
+from _typeshed import SupportsWrite
+from contextlib import _GeneratorContextManager
+from types import TracebackType
+from typing import Any, Final, Literal, Protocol, TextIO, TypeVar, overload
+from typing_extensions import TypeAlias
+
+STDOUT: Final = 2
+PIPE: Final = 3
+_STDOUT: TypeAlias = Literal[2]
+_PIPE: TypeAlias = Literal[3]
+_T_contra = TypeVar("_T_contra", contravariant=True)
+_StreamOutT = TypeVar("_StreamOutT", bound=_Stream[str] | _Stream[bytes])
+_StreamErrT = TypeVar("_StreamErrT", bound=_Stream[str] | _Stream[bytes])
+
+class _Stream(SupportsWrite[_T_contra], Protocol):
+    def seek(self, __offset: int, __whence: int = ...) -> int: ...
+
+# Alias for IPython.core.interactiveshell.InteractiveShell.
+# N.B. Even if we added ipython to the stub-uploader allowlist,
+# we wouldn't be able to declare a dependency on ipython here,
+# since `wurlitzer` does not declare a dependency on `ipython` at runtime
+_InteractiveShell: TypeAlias = Any
+
+class Wurlitzer:
+    flush_interval: float = 0.2
+
+    def __init__(
+        self,
+        stdout: _Stream[str] | _Stream[bytes] | None = None,
+        stderr: _STDOUT | _Stream[str] | _Stream[bytes] | None = None,
+        encoding: str | None = ...,
+        bufsize: int | None = ...,
+    ) -> None: ...
+    def __enter__(self): ...
+    def __exit__(
+        self, exc_type: type[BaseException] | None, exc_value: BaseException | None, traceback: TracebackType | None
+    ) -> None: ...
+
+def dup2(a: int, b: int, timeout: float = 3) -> int: ...
+def sys_pipes(encoding: str = ..., bufsize: int | None = None) -> contextlib._GeneratorContextManager[tuple[TextIO, TextIO]]: ...
+
+# stubtest does not support overloaded context managers, hence the _GeneratorContextManager[Foo] return types.
+@overload
+def pipes(
+    stdout: _PIPE, stderr: _STDOUT, encoding: None, bufsize: int | None = None
+) -> _GeneratorContextManager[tuple[io.BytesIO, None]]: ...
+@overload
+def pipes(
+    stdout: _PIPE, stderr: _PIPE, encoding: None, bufsize: int | None = None
+) -> _GeneratorContextManager[tuple[io.BytesIO, io.BytesIO]]: ...
+@overload
+def pipes(
+    stdout: _PIPE, stderr: _StreamErrT, encoding: None, bufsize: int | None = None
+) -> _GeneratorContextManager[tuple[io.BytesIO, _StreamErrT]]: ...
+@overload
+def pipes(
+    stdout: _PIPE, stderr: _STDOUT, encoding: str, bufsize: int | None = None
+) -> _GeneratorContextManager[tuple[io.StringIO, None]]: ...
+@overload
+def pipes(
+    stdout: _PIPE, stderr: _PIPE, encoding: str, bufsize: int | None = None
+) -> _GeneratorContextManager[tuple[io.StringIO, io.StringIO]]: ...
+@overload
+def pipes(
+    stdout: _PIPE, stderr: _StreamErrT, encoding: str, bufsize: int | None = None
```

Comment:
The encoding is a string by default, so `encoding: str` overloads should instead be `encoding: str = ...`. This way they are used when no encoding is given, and wurlitzer uses `"utf-8"` or whatever the default happens to be.

```suggestion
    stdout: _PIPE, stderr: _STDOUT, encoding: str = ..., bufsize: int | None = None
) -> _GeneratorContextManager[tuple[io.StringIO, None]]: ...
@overload
def pipes(
    stdout: _PIPE, stderr: _PIPE, encoding: str = ..., bufsize: int | None = None
) -> _GeneratorContextManager[tuple[io.StringIO, io.StringIO]]: ...
@overload
def pipes(
    stdout: _PIPE, stderr: _StreamErrT, encoding: str = ..., bufsize: int | None = None
```